multiversx_sdk/core/transaction_computer.py

```diff
@@ -41,6 +41,19 @@ def compute_transaction_fee(self, transaction: Transaction, network_config: INet
         return int(fee_for_move + processing_fee)
 
     def compute_bytes_for_signing(self, transaction: Transaction) -> bytes:
+        """Computes the bytes for signing based on the `version` and `options` of the transaction.
+        If the least significant bit of the `options` is set, will serialize transaction for hash signing."""
+        self._ensure_fields(transaction)
+
+        if transaction.version >= MIN_TRANSACTION_VERSION_THAT_SUPPORTS_OPTIONS:
+            if self.has_options_set_for_hash_signing(transaction):
+                return self.compute_hash_for_signing(transaction)
+
+        dictionary = self._to_dictionary(transaction)
+        serialized = self._dict_to_json(dictionary)
+        return serialized
+
+    def compute_bytes_for_ledger_signing(self, transaction: Transaction) -> bytes:
```

Comment:
I think we should not have the term "ledger" in the transaction computer (a somehow core component). Maybe incorporate the term "signing kind / type"?

Some examples:
 - the new function can be called `compute_bytes_for_signing_by_ignoring_transaction_options`
 - on the exiting function, we can add a new parameter e.g. `ignore_options = False`.

Reply:
Once we decide, we should also update the specs :pray:

Reply:
refactored it